crypto.go

```diff
@@ -0,0 +1,50 @@
+package srtp
+
+import (
+	"crypto/cipher"
+)
+
+// xorBytes computes the exclusive-or of src1 and src2 and stores it in dst.
+// It returns the number of bytes written.
+func xorBytes(dst, src1, src2 []byte) int {
+	n := len(src1)
+	if len(src2) < n {
+		n = len(src2)
+	}
+	if len(dst) < n {
+		n = len(dst)
+	}
+
+	for i := 0; i < n; i++ {
+		dst[i] = src1[i] ^ src2[i]
+	}
+
+	return n
+}
+
+// incrementCTR increments a big-endian integer of arbitrary size.
+func incrementCTR(ctr []byte) {
+	for i := len(ctr) - 1; i >= 0; i-- {
+		ctr[i]++
+		if ctr[i] != 0 {
+			break
+		}
+	}
+}
+
+// xorBytesCTR performs CTR encryption and decryption.
+// It is equivalent to cipher.NewCTR followed by XORKeyStream.
+func xorBytesCTR(block cipher.Block, iv []byte, dst, src []byte) {
```

Comment:
Please add following error handlings to avoid future security problems. Having tests for error cases is better for the security.
- initial vector length, input buffer sizes, and block size
- validity of overwrapped buffer address (ones in the official crypto package: https://github.com/golang/go/blob/50bd1c4d4eb4fac8ddeb5f063c099daccfb71b26/src/crypto/cipher/ctr.go#L79-L81)

Reply:
* iv length: done;
* input buffer size: this is CTR, it works with arbitrary sizes;
* block size: this is handled by `NewCipher`, which is not part of my code;
* overlap: I cannot do that without importing `unsafe`, which I'm hesitant to do.